rust/hyperlane-core/src/config/mod.rs

```diff
@@ -122,3 +123,33 @@ impl Display for ConfigParsingError {
 }
 
 impl std::error::Error for ConfigParsingError {}
+
+/// Try to unwrap a series of options during config parsing and handle errors more gracefully than
+/// unwrapping and causing a panic if we forgot to assert something earlier.
+///
+/// Use as `cfg_unwrap_all!(cwp, err: a, b, c)` where `cwp` is the current working path and `err` is
+/// the `ConfigParsingError`, and a, b, and c are the options to unwrap. The result will be that
+/// calling this macro a, b, and c will be unwrapped and assigned to variables of the same name.
+#[macro_export]
+macro_rules! cfg_unwrap_all {
+    ($cwp:ident, $err:ident: $($i:ident),+$(,)?) => {
+        #[allow(unused_parens)]
+        let ($($i),*) = if let ($(Some($i)),*) = ($($i),*) {
+            ($($i),*)
+        } else if $err.is_ok() {
+            $err.push($cwp.clone(), eyre::eyre!(
+                    "Invalid configuration; seeing this error means we forgot to handle a specific error case before unwrapping it. Occurred when accessing ({})",
```

Comment:
Since this `else if` branch only prints an additional message compared to the `else` branch below, I think we could remove the duplication and also possibly reduce confusion about why the first logged parsing error has a slightly different error.

What do you think about making the `ConfigPath` in `ConfigParsingError` optional so that `"Invalid configuration; seeing this error means we forgot to handle a specific error case before unwrapping it."` could be passed in without a path and then just use the logic in the `else branch`?

Reply:
Pushed something along these lines.